crates/vm/levm/src/opcode_handlers/environment.rs

```diff
@@ -194,22 +194,12 @@ impl VM {
         &mut self,
         current_call_frame: &mut CallFrame,
     ) -> Result<OpcodeSuccess, VMError> {
-        if self
-            .env
-            .consumed_gas
-            .checked_add(gas_cost::CODESIZE)
-            .ok_or(VMError::OutOfGas(OutOfGasError::ConsumedGasOverflow))?
-            > self.env.gas_limit
-        {
-            return Err(VMError::OutOfGas(OutOfGasError::MaxGasLimitExceeded));
-        }
```

Comment:
@JereSalo is this check made in an upper layer?

Reply:
IINM the `increase_consumed_gas` function  already does that checking. 

Source: 
https://github.com/lambdaclass/ethrex/blob/main/crates/vm/levm/src/vm.rs#L1010-L1023

So, in theory, by calling the `increase_consumed_gas` before anything, that check is done.

Reply:
You're right!